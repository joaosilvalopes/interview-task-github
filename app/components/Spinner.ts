import styled from "styled-components";

const Spinner = styled.div`
    width: 1em;
    height: 1em;
    border: 3px solid transparent;
    border-radius: 50%;
    border-top-color: currentColor;
    border-right-color: currentColor;
    animation: spin 1s ease-out infinite;
  }
  
  @keyframes spin {
    to { transform: rotate(360deg); }
  }
`;

export default Spinner;