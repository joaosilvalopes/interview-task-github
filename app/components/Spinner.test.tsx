import { render } from '@testing-library/react';

import Spinner from './Spinner';

describe('Spinner component', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Spinner />);

    expect(asFragment()).toMatchSnapshot();
  });
});
