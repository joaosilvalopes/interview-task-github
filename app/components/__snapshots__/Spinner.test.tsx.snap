// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-beySbM iwRbkc"
  />
</DocumentFragment>
`;
